### DISCUSSION TITLE
Problema con Shelly EM 01

### DISCUSSION THREAD

**Discussion by barbatrukko** [indomus/forum#687] [HUB: Home Assistant]
Salve,
ho montato uno shelly EM 01 sul mio impianto per monitorare alcuni consumi.
Per un po' ha funzionato ma ora mi trovo che da un po' di tempo Home Assistant ogni tanto non legge i dati corretti.
Ad esempio controllando stamattina vedo sull'app di shelly i valori corretti in tempo reale dei due canali (e se apro e spengo luci vedo in tempo reale il consumo modificarsi), mentre su Home Assistant vedo i valori "congelati" a ieri sera alle 23 circa che sono ripetuti nel grafico fino a tutt'ora.
Ovviamente questo inciderà anche su eventuali calcoli di consumi giornalieri, settimanali, etc.
Ho effettuato il riavvio ed ora anche HA legge i due valori in tempo reale in maniera corretta.
Ma io non potrò mai sapere se nel tempo ha già fatto questo scherzetto.... Come faccio ad essere sicuro dei dati rilevati?
Tra ieri e oggi so che per 12 ore ha "segnato" 124W invece di 4W... quindi il dato è alquanto sfalsato.

**Comment by essegi65** [comment]
Purtroppo non mi dici niente di nuovo ho aperto anche io alcune discussioni su la reale lettura del mio shelly em. Ad oggi non ne sono venuto a capo, io sto usando l'integrazione originale di HA prossimamente proverò a mettere tutto sotto MQTT e togliere l'integrazione per vedere se cambia qualcosa. Le mie letture non reagiscono come le tue ma mi trovo dei picchi di consumo che arrivano (secondo le letture dello shelly) a 6kw quando il mio contatore è tarato a 3kw, quindi impossibili da raggiungere 6kw.

**Comment by gsarcl366** [comment]
Ho notato che bisogna aggiornare costantemente gli shelly altrimenti non sempre rispondono per il resto sono ottimi dispositivi.

**Reply by snakuzzo** [reply]
che intendi dire con "aggiornare" ?

**Reply by gsarcl366** [reply]
L'app prima di tutto e poi eventuali aggiornamenti di firmware

**Reply by snakuzzo** [reply]
l'app per il mio utilizzo è inutile.
Quanto al discorso degli aggiornamenti firmware invece, non credo proprio sia così.
Hanno ultimamente risolto (dicono) i problemi di connettività legati alla gestione del WIFI perché hanno utilizzato il nuovo SDK di Espressif, ma non è che se non aggiorni il firmware poi i dispositivi non funzionano....e questo vale soprattutto se li integri via MQTT

**Reply by essegi65** [reply]
Condivido quanto detto da @snakuzzo, io lo tengo sempre aggiornato all'ultimo firmware e le letture non si modificano.
Piuttosto, devo approfondire, ma le letture dello shelly mi sembra variano a seconda della versione di hassio che ho, ancora non ho ben capito il motivo però (potrebbe essere un qualcosa legato all'integrazione?). Vedremo.......

**Comment by barbatrukko** [comment]
Io aggiorno sempre tutto, app, firmware, HomeAssistant.
Mi sono accorto per sbaglio in questo caso che HA ha riportato per 12 ore lo stesso valore (diverso da quello che leggevo in tempo reale sull'APP).
alla fine ho dovuto riavviare HA ed al riavvio ha ripreso a leggere il valore corretto....
Di fatto me ne sono accorto per caso, ma a questo punto se dovessi considerare il conteggio che ha fatto HA per il mese...avrei i miei dubbi.

Io credo che questi dispositivi dovrebbero avere un loro contatore "interno" che a prescindere dala connessione ad internet o al gateway tenga sempre il conteggio dei consumi, in modo che anche se non riesci a contattare il dispositivo per 10giorni, a fine mese comunque hai la lettura del totale da confrontare con i tuoi conteggi derivati dal realtime (comodo per i grafici ma poco pratico per l'affidabilità)

**Reply by NdK73** [reply]
Con l'ultimo aggiornamento l'integrazione Shelly mi pare aver perso qualità. Senza arrivare alle 12 ore, ma ho dei dimmer che prima funzionavano bene ed ora invece si aggiornano solo col polling. Cioè, se da GUI di HA accendo la luce, dopo un attimo l'interruttore torna in 'off', per tornare 'on' al polling successivo. Prima la reazione era immediata. Ovviamente connessione solo locale, senza cloud. E pare che non abbiano ancora aggiornato i profili per il 'nuovo' sistema di misura dell'energia (bisogna ancora applicare la 'patch' nella configurazione perché permetta di aggiungere lo shelly EM come misuratore di produzione e consumo). E' un peccato perché i prodotti sono veramente validi...

**Comment by barbatrukko** [comment]
> Con l'ultimo aggiornamento l'integrazione Shelly mi pare aver perso qualità. Senza arrivare alle 12 ore, ma ho dei dimmer che prima funzionavano bene ed ora invece si aggiornano solo col polling. Cioè, se da GUI di HA accendo la luce, dopo un attimo l'interruttore torna in 'off', per tornare 'on' al polling successivo. Prima la reazione era immediata. Ovviamente connessione solo locale, senza cloud. E pare che non abbiano ancora aggiornato i profili per il 'nuovo' sistema di misura dell'energia (bisogna ancora applicare la 'patch' nella configurazione perché permetta di aggiungere lo shelly EM come misuratore di produzione e consumo). E' un peccato perché i prodotti sono veramente validi...

Mi sai dare info sulla patch? che patch è ? Anche io ho lo stesso problema e ne meross, be shelly vengono visualizzati nella sezione misurazione

**Comment by NdK73** [comment]
L'ho trovata su un altro sito (che non si può linkare). Comunque è molto semplice:
```
homeassistant:
  customize_glob:
    "sensor.shelly_*_total_consumption":
      last_reset: '1970-01-01T00:00:00+00:00'
      state_class: total_increasing
      device_class: energy
```
Purtroppo, se invece di un 'total consumption'' in kWh si ha solo il consumo (o la produzione) attuale in kW, non ho ancora capito come farglielo vedere :(

**Reply by snakuzzo** [reply]
Non c'era bisogno visto che le indicazioni le abbiamo date pure noi...e soprattutto mi sa che il last_reset non è più supportato

**Reply by NdK73** [reply]
Possibile. Però non ho visto errori, e pare funzionare. Almeno per ora :)
Forse lo hanno deprecato ma è ancora accettato?

Intanto ho trovato che per poter usare nel pannello "energia" i dati prelevati da un misuratore di consumi che fornisce solo valori istantanei va creto un sensore virtuale (o due, in e out), tipo:
```
sensor:
  - platform: integration
    name: EnergiaIn
    source: sensor.fv_import
    method: left
```
